crates/proto/src/rr/domain/name.rs

```diff
@@ -718,6 +719,45 @@ impl Name {
         format!("{}", self)
     }
 
+    /// Converts a *.arpa Name in a PTR record back into an IpNet if possible.
+    pub fn parse_arpa_name(&self) -> Result<IpNet, ProtoError> {
+        let mut iter = self.iter().rev();
+        if !"arpa".eq_ignore_ascii_case(std::str::from_utf8(iter.next().ok_or(Err("not an arpa address".into()))?)?) {
+            return Err("not an arpa address".into());
+        }
+        let mut prefix_len: u8 = 0;
+        match &std::str::from_utf8(iter.next().ok_or(Err("invalid arpa address".into()))?)?.to_ascii_lowercase()[..] {
+            "in-addr" => {
+                let mut octets: [u8; 4] = [0; 4];
+                for octet in octets.iter_mut() {
+                    match iter.next() {
+                        Some(label) => *octet = std::str::from_utf8(label)?.parse()?,
+                        None => break,
+                    }
+                    prefix_len += 8;
+                }
+                Ok(IpNet::V4(Ipv4Net::new(octets.into(), prefix_len).expect("Ipv4Net::new")))
```

Comment:
Any reason you're pulling in a new dependency here? This looks almost exactly like the `std::net::IpAddr` constructor, not sure I see the reason to add a dependency for that.

Reply:
We need to store `prefix_len`.

Reply:
Ah I missed that.

Reply:
Ok, I just quickly reviewed that library. I don't see anything majorly concerning there.